cli/src/main/java/org/jboss/sbomer/cli/commands/processor/RedHatProductProcessCommand.java

```diff
@@ -0,0 +1,114 @@
+import org.cyclonedx.model.Component;
+import org.jboss.pnc.dto.Build;
+import org.jboss.pnc.dto.BuildConfiguration;
+import org.jboss.sbomer.cli.commands.processor.ProductVersionMapper.ProductVersionMapping;
+import org.jboss.sbomer.cli.model.Sbom;
+import org.jboss.sbomer.core.enums.ProcessorImplementation;
+import org.jboss.sbomer.core.errors.ApplicationException;
+import org.jboss.sbomer.core.utils.SbomUtils;
+
+import lombok.extern.slf4j.Slf4j;
+import picocli.CommandLine.Command;
+
+/**
+ * <p>
+ * Processor to add the Red Hat Product information to the main {@link Component} within the {@link Bom}.
+ * </p>
+ */
+@Slf4j
+@Command(
+        mixinStandardHelpOptions = true,
+        name = "redhat-product",
+        description = "Process the SBOM with Red Hat product enrichment")
+public class RedHatProductProcessCommand extends AbstractProcessCommand {
+
+    public final static String PROPERTY_ERRATA_PRODUCT_NAME = "errata-tool-product-name";
+    public final static String PROPERTY_ERRATA_PRODUCT_VERSION = "errata-tool-product-version";
+    public final static String PROPERTY_ERRATA_PRODUCT_VARIANT = "errata-tool-product-variant";
+
+    @Inject
+    ProductVersionMapper productVersionMapper;
+
+    /**
+     * <p>
+     * Adds a property of a given name in case it's not already there.
+     * </p>
+     *
+     * @param component The {@link Component} to add the property to.
+     * @param property The name of the property.
+     * @param value The value of the property.
+     */
+    private void addPropertyIfMissing(Component component, String property, String value) {
+        if (!SbomUtils.hasProperty(component, property)) {
+            log.debug("Adding {} property with value: {}", property, value);
+            SbomUtils.addProperty(component, property, value);
+        } else {
+            log.debug(
+                    "Property {} already exist, value: {}",
+                    property,
+                    SbomUtils.findPropertyWithNameInComponent(property, component).get().getValue());
+        }
+    }
+
+    @Override
+    public ProcessorImplementation getImplementationType() {
+        return ProcessorImplementation.REDHAT_PRODUCT;
+    }
+
+    @Override
+    public Bom doProcess(Sbom sbom) {
+        Bom bom = getBom(sbom);
+        Build build = pncService.getBuild(sbom.getBuildId());
+
+        if (build == null) {
+            throw new ApplicationException(
+                    "Build related to the SBOM could not be found in PNC, interrupting processing");
+        }
+
+        BuildConfiguration buildConfig = pncService.getBuildConfig(build.getBuildConfigRevision().getId());
+
+        if (buildConfig == null) {
+            throw new ApplicationException(
+                    "BuildConfig related to the SBOM could not be found in PNC, interrupting processing");
+        }
+
+        ProductVersionMapping mapping = productVersionMapper.getMapping().get(buildConfig.getProductVersion().getId());
```

Comment:
ProductVersion might be null in the BuildConfigs

Reply:
Thanks, will update it!

Reply:
This should be now covered!